### PR TITLE
expand the CVO featureset annotation to include other FeatureSets

### PR DESCRIPTION
**Pull request by deads2k**
expand the enhancement to include the details from https://github.com/openshift/cluster-version-operator/pull/821

/assign @wking

### CODE REVIEW

File: enhancements/update/cvo-techpreview-manifests.md [hunk 1]
```diff
@@ -52,28 +53,33 @@ resources like namespaces, roles, rolebindings, serviceacounts, and deployments.
...
-5. Allowing manifests condition on other featuresets in this iteration.
 
 ## Proposal
 
-Add an annotation that can be set on CVO manifests called `release.openshift.io/feature-gate` with one legal/honored value
-of "TechPreviewNoUpgrade".
-If the value is not "TechPreviewNoUpgrade", the manifest is never created.
+Add an annotation that can be set on CVO manifests called `release.openshift.io/feature-set` that can be set to a comma
+delimited list that can contain any value from 
+[FeatureSet](https://github.com/openshift/api/blob/master/config/v1/types_feature.go#L28-L43) and "Default" in place of "".
```
Comment (wking) [inline enhancements/update/cvo-techpreview-manifests.md:61]: nit: line references like `#L28-L43` mixed with a floating ref like `master` are brittle.  Can we use [this][1] instead?

[1]: https://pkg.go.dev/github.com/openshift/api/config/v1#FeatureSet

File: enhancements/update/cvo-techpreview-manifests.md [hunk 2]
```diff
@@ -52,28 +53,33 @@ resources like namespaces, roles, rolebindings, serviceacounts, and deployments.
...
+not be created unless `featuregates.config.openshift.io|.spec.featureSet=""` (empty string).
+If a manifest sets `.metadata.annotations["release.openshift.io/feature-set"]="Default,LatencySensitive"`, the manifest will
+not be created unless `featuregates.config.openshift.io|.spec.featureSet` is `""` or `"LatencySensitive"`.
 During bootstrapping, the CVO will assume no feature sets are enabled until it can successfully retrieve
 `featuregates.config.openshift.io` from the Kubernetes API server.
 If bootstrapping changes are required, it will be responsibility of the contributing team to manage the proper inclusion
 in the installer and other bootstrapping components.
 
-Because the cluster cannot upgrade to 4.y+1 and because the TechPreviewNoUpgrade flag cannot be unset, there is no concern
-about removing any manifests that were added for TechPreviewNoUpgrade.
```
Comment (wking) [inline enhancements/update/cvo-techpreview-manifests.md:76]: `Version Skew Strategy` section will also need an overhaul.

**Comment by deads2k** [comment]
updated for comments.

**Review by wking — APPROVED** [review]
/lgtm

**Pull request merged**